src/lbfgs.jl

```diff
@@ -89,10 +89,16 @@ function LBFGSSolver(nlp::M; mem::Int = 5) where {T, V, M <: AbstractNLPModel{T,
   return LBFGSSolver{T, V, Op, M}(x, xt, gx, gt, d, H, h)
 end
 
-function LinearOperators.reset!(solver::LBFGSSolver)
+function SolverCore.reset!(solver::LBFGSSolver)
   reset!(solver.H)
 end
 
+function SolverCore.reset!(solver::LBFGSSolver, nlp::AbstractNLPModel{T, V}) where {T, V}
```

Comment:
```suggestion
function SolverCore.reset!(solver::LBFGSSolver, nlp::AbstractNLPModel)
```

Reply:
Do you want `solver` and `nlp` of the same type? ({T, V})
If it's the case you need to parameterize both, otherwise if you don't need parameters `{T, V}`, you can remove.
The compiler will have less pressure and it will not compile a new method for each `T` or `V`.